## Zadatak 1

In [21]:
m = 2137
a = 1

for j in range(m - 3):
    a+=1
    nasao = 0
    for i in range(m):
        if (i * i - a) % m == 0:
            print( a, i, sep=" " )
            nasao = 1
            break
    if nasao == 0:
        print(a)
        break

2 364
3 384
4 2
5


Traženo rješenje je **5**.

## Zadatak 2

In [19]:
p = 2137
a = 1902
d = 5
s = 3
t = 267
print( 2 ** 3 * 267, p - 1, sep=" = ")

A = (a ** t) % p
D = (d ** t) % p

m = 0

for i in range(s):
    if ( (A*(D**m))**(2**(s - 1 - i)) + 1) % p == 0:
        m = m + 2 ** i
x = ( (a ** int((t+1)/2)) * (D ** int(m/2)) ) % p

print( "Rjesenje: ", x )

2136 = 2136
Rjesenje:  1804


Rješenja su: $x \equiv 333$ (mod 2137), $x \equiv 1804$ (mod 2137).

## Zadatak 3

Uzmimo Q = 2000 i formiramo matricu B kao iz teorema 1.5 i pomoću PARI-ja izračunamo <code>qflll(B)</code>. Iz prvog stupca dobivene matrice pročitamo brojeve $q = 1534858$, $p_1 = 2170617$, $p_2 = 11789468$. Pomoću WolframAlpha-e provjerimo da dobiveni brojevi zadovoljavaju uvjete zadatka.